receiver/collectdreceiver/observability.go

```diff
@@ -6,79 +6,66 @@ package collectdreceiver // import "github.com/open-telemetry/opentelemetry-coll
 import (
 	"context"
 
-	"go.opencensus.io/stats"
-	"go.opencensus.io/stats/view"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/otel/metric"
 )
 
-func init() {
-	_ = view.Register(
-		viewInvalidRequests,
-		viewRequestsReceived,
-		viewMetricsReceived,
-		viewEventsReceived,
-		viewBlankDefaultAttrs,
-	)
+type observability struct {
+	mErrors            metric.Int64Counter
+	mRequestsReceived  metric.Int64Counter
+	mMetricsReceived   metric.Int64Counter
+	mEventsReceived    metric.Int64Counter
+	mBlankDefaultAttrs metric.Int64Counter
 }
 
-var (
-	mErrors            = stats.Int64("otelcol/collectd/errors", "Errors encountered during processing of collectd requests", "1")
-	mRequestsReceived  = stats.Int64("otelcol/collectd/requests_received", "Number of total requests received", "1")
-	mMetricsReceived   = stats.Int64("otelcol/collectd/metrics_received", "Number of metrics received", "1")
-	mEventsReceived    = stats.Int64("otelcol/collectd/events_received", "Number of events received", "1")
-	mBlankDefaultAttrs = stats.Int64("otelcol/collectd/blank_default_attrs", "Number of blank default attributes received", "1")
-)
-
-var viewInvalidRequests = &view.View{
-	Name:        mErrors.Name(),
-	Description: mErrors.Description(),
-	Measure:     mErrors,
-	Aggregation: view.Sum(),
-}
-
-var viewRequestsReceived = &view.View{
-	Name:        mRequestsReceived.Name(),
-	Description: mRequestsReceived.Description(),
-	Measure:     mRequestsReceived,
-	Aggregation: view.Sum(),
-}
-
-var viewMetricsReceived = &view.View{
-	Name:        mMetricsReceived.Name(),
-	Description: mMetricsReceived.Description(),
-	Measure:     mMetricsReceived,
-	Aggregation: view.Sum(),
-}
-
-var viewEventsReceived = &view.View{
-	Name:        mEventsReceived.Name(),
-	Description: mEventsReceived.Description(),
-	Measure:     mEventsReceived,
-	Aggregation: view.Sum(),
-}
+func initMetrics(telemetrySettings component.TelemetrySettings) (*observability, error) {
+	meter := telemetrySettings.MeterProvider.Meter("collectd")
 
-var viewBlankDefaultAttrs = &view.View{
-	Name:        mBlankDefaultAttrs.Name(),
-	Description: mBlankDefaultAttrs.Description(),
-	Measure:     mBlankDefaultAttrs,
-	Aggregation: view.Sum(),
+	mErrors, err := meter.Int64Counter("errors", metric.WithUnit("1"), metric.WithDescription("Errors encountered during processing of collectd requests"))
```

Comment:
is the metric name preserved with this change?

Reply:
No, in all likelihood I broke this.

Reply:
I meant to change those names anyway. I don't think "otel/collectd/errors" was a good name. I'm going to look for ways to also make those metrics optional by default. So we're going for a breaking change here.